simavr/sim/avr_uart.c

```diff
@@ -532,8 +532,11 @@ avr_uart_init(
 	// status bits
 	// monitor code that reads the rxc flag, and delay it a bit
 	avr_register_io_read(avr, p->rxc.raised.reg, avr_uart_status_read, p);
-	if (p->fe.reg != p->rxc.raised.reg)
-		avr_register_io_read(avr, p->fe.reg, avr_uart_status_read, p);
+    if (p->fe.reg) {
```

Comment:
As this test evaluates `p->fe.reg`, it implicitly assumes this expression is well defined. But if `p->fe.reg` is well defined, then what is the purpose of the test?